pkg/oauth/registry/oauthauthorizetoken/etcd/etcd.go

```diff
@@ -0,0 +1,55 @@
+package etcd
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+
+	"github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/oauthauthorizetoken"
+)
+
+// rest implements a RESTStorage for authorize tokens against etcd
+type REST struct {
+	etcdgeneric.Etcd
+}
+
+const EtcdPrefix = "/registry/oauth/authorizetokens"
+
+// NewREST returns a RESTStorage object that will work against authorize tokens
+func NewREST(h tools.EtcdHelper) *REST {
+	store := &etcdgeneric.Etcd{
+		NewFunc:     func() runtime.Object { return &api.OAuthAuthorizeToken{} },
+		NewListFunc: func() runtime.Object { return &api.OAuthAuthorizeTokenList{} },
+		KeyRootFunc: func(ctx kapi.Context) string {
+			// TODO: JTL: switch to NoNamespaceKeyRootFunc after rebase
+			return EtcdPrefix
+		},
+		KeyFunc: func(ctx kapi.Context, name string) (string, error) {
+			return etcdgeneric.NoNamespaceKeyFunc(ctx, EtcdPrefix, name)
+		},
+		ObjectNameFunc: func(obj runtime.Object) (string, error) {
+			return obj.(*api.OAuthAuthorizeToken).Name, nil
+		},
+		PredicateFunc: func(label labels.Selector, field fields.Selector) generic.Matcher {
+			return oauthauthorizetoken.Matcher(label, field)
+		},
+		EndpointName: "oauthaccesstokens",
```

Comment:
fixed... looks like it's only used for error reporting, which is why tests didn't complain